### ISSUE TITLE
Feature Request: Allow double clicking the tray icon to open the 'nextcloud' folder

### ISSUE THREAD

**Issue by DeMoB** [nextcloud/desktop#1184]
Dropbox allows double clicking the application's tray icon to open the main sync folder.

I'd love if a similar feature could be implemented in nextcloud.

Single click could still be left as opening the app status/settings, but the double click captured to open the folder.

**Comment by michaelstingl** [comment]
> Dropbox allows double clicking the application's tray icon to open the main sync folder.

Not on macOS. (v69.4.102)

**Comment by alescdb** [comment]
I would also love to have this, It make more sense to open the nextcloud folder that the activity window by defaut on double click, especialy for those like me that are switching from Dropbox to Nextcloud (at least on Windows and Linux).
Maybe this can be an option in the settings?

**Comment by alescdb** [comment]
If someone is interested, I've made a simple, quick & dirty patch (works with current master).
You can apply it with :

```git apply nc.patch```

and compile like usual (cf. `REAMDE.md`).

```
diff --git a/src/gui/owncloudgui.cpp b/src/gui/owncloudgui.cpp
index 53b568413..40f572abc 100644
--- a/src/gui/owncloudgui.cpp
+++ b/src/gui/owncloudgui.cpp
@@ -147,6 +147,16 @@ void ownCloudGui::slotOpenSettingsDialog()
     }
 }
 
+bool ownCloudGui::openFirstFolder()
+{
+    FolderMan *folderMan = FolderMan::instance();
+    if (folderMan->map().size() > 0) {
+        this->slotFolderOpenAction(folderMan->map().first()->alias());
+        return (true);
+    }
+    return (false);
+}
+
 void ownCloudGui::slotTrayClicked(QSystemTrayIcon::ActivationReason reason)
 {
     if (_workaroundFakeDoubleClick) {
@@ -176,6 +186,9 @@ void ownCloudGui::slotTrayClicked(QSystemTrayIcon::ActivationReason reason)
                 raiseDialog(_settingsDialog.data());
             }
 #else
+            if (this->openFirstFolder()) {
+                return;
+            }
             slotOpenSettingsDialog();
 #endif
         }
diff --git a/src/gui/owncloudgui.h b/src/gui/owncloudgui.h
index 105235233..566711d43 100644
--- a/src/gui/owncloudgui.h
+++ b/src/gui/owncloudgui.h
@@ -100,6 +100,7 @@ public slots:
     void slotTrayMessageIfServerUnsupported(Account *account);
     void slotNavigationAppsFetched(const QJsonDocument &reply, int statusCode);
     void slotEtagResponseHeaderReceived(const QByteArray &value, int statusCode);
+    bool openFirstFolder();
 
 
     /**
```

**Comment by brainchild0** [comment]
> *Single click could still be left as opening the app status/settings*, but the double click captured to open the folder.

On my system double clicking causes no specific or helpful action.  I observe that single clicking opens (or closes) the drop down menu, and double clicking is processed as two single clicks.  Opening the settings window is impossible except through using the menu entry, which is inconvenient as it is by far the most common action.

Perhaps  it is a platform issue.

I am running Linux Mint 19.2 with Cinnamon Desktop.

NC Client version is 2.6.0.

**Comment by bcurran3** [comment]
I would give my left, um, ...every GitHub + point I have for this feature on Windows.

**Comment by er-vin** [comment]
@jancborchardt opinion there?

**Comment by jancborchardt** [comment]
If this is something that Google Drive and Dropbox (still) do, then it would be nice to have for sure. Even if it seems to be only a thing on Windows, still nice to have it there for people who are used to it.

**Comment by DeMoB** [comment]
I don't have Google Drive installed to check, but Dropbox and OneDrive still have this behaviour on windows.

**Comment by bcurran3** [comment]
Google Backup and Sync (formerly Google Drive) has this behavior. Double clicking the system tray icon opens up the local storage folder for the program.

**Comment by er-vin** [comment]
> If this is something that Google Drive and Dropbox (still) do, then it would be nice to have for sure. Even if it seems to be only a thing on Windows, still nice to have it there for people who are used to it.

I just wonder which folder it should open though, you can have several accounts with several sync folders for each. And their orders is not really configurable by the user so picking "the first one we find" is a bit like picking at random.

**Comment by alescdb** [comment]
@er-vin If the "double click to open folder" is made optional, a sub-option may be added to select the default account/folder to open ?!

**Comment by jancborchardt** [comment]
> I just wonder which folder it should open though, you can have several accounts with several sync folders for each. And their orders is not really configurable by the user so picking "the first one we find" is a bit like picking at random.

Good point. For the simplest case with 1 account and 1 folder, we can open that one. But for additional ones it’s indeed a question.
@alescdb this would not be optional but just work, and sub-options for this would be overdoing it. This should simply work and not require additional config.

@DeMoB Dropbox and OneDrive also have multi-account support, do they? How is it handled there?

**Comment by DeMoB** [comment]
> @DeMoB Dropbox and OneDrive also have multi-account support, do they? How is it handled there?
While I only have a free tier Dropbox account, I can't see an option to add a second account on there.

Google Backup & Sync and OneDrive appear to create a new instance of the system tray app for each account you have signed in.

**Comment by foorschtbar** [comment]
Any update? I have switched from Dropbox to Nextcloud and I really miss this behavior...

**Comment by er-vin** [comment]
Waiting on @jancborchardt opinion on how to deal with the multi-account case.

**Comment by jancborchardt** [comment]
Let’s do it simply:
- For the simplest case with 1 account and 1 folder, we can open that one.
- When multiple accounts are configured, double clicking the tray icon will open the folder of the account which is currently/last selected in the account switcher of the tray dialog.

-----------

> Google Backup & Sync and OneDrive appear to create a new instance of the system tray app for each account you have signed in.

Oh, okay? I’m not sure if this is something we want to do, as even already with 2–3 icons (which will have no text as they are in the systray) will make it super confusing what relates to which account. Can you show a screenshot @DeMoB?

**Comment by DeMoB** [comment]
> Oh, okay? I’m not sure if this is something we want to do, as even already with 2–3 icons (which will have no text as they are in the systray) will make it super confusing what relates to which account. Can you show a screenshot @DeMoB?

I only installed them for testing, so don't have them anymore but the icons were identical.

You could only distinguish between the accounts via the hover tooltip text on the system tray icon (e.g.. Onedrive says "OneDrive - [Account Name], [Sync Status]"), or opening the pop-up menus from the tray icon.

**Comment by er-vin** [comment]
> Let’s do it simply:
> 
>     * For the simplest case with 1 account and 1 folder, we can open that one.
> 
>     * When multiple accounts are configured, double clicking the tray icon will open the folder of the account which is currently/last selected in the account switcher of the tray dialog.

Let's go for that then :+1:

**Comment by biva** [comment]
In my opinion, it would be more useful to access the Nextcloud setting window when I double click on it. 

Indeed, if I need to access my Nextcloud folder easily, I just have to create a shortcut in the taskbar. If I have several accounts synced, I can create as much shortcut as I need, without creating confusion with the multi-account case mentioned above.

While if I need to access the Nextcloud setting (for example to force the sync of a synced folder, or to check which folder has an issue, ...), I need to click right and select Settings. It would also solve the question: "how to deal with the multi-account case".

**Comment by skjerns** [comment]
Nice to see this on the roadmap. This could also be something that can be changed in the settings. For example: In Dropbox you can select whether double clicking opens the Dropbox app or the Explorer.

**Comment by jancborchardt** [comment]
This is a rather simple enhancement it seems, and we have a spec at https://github.com/nextcloud/desktop/issues/1184#issuecomment-690140761

What do you think @claucambra @mgallien? :)
